File: chroma-league/src/test/java/net/booone/chromaleague/hud/animations/ActivePlayerKillAnimationTest.java
package net.booone.chromaleague.hud.animations;

import net.booone.chromaleague.GameStateMocks;
import net.booone.chromaleague.hud.AnimationTester;
import org.junit.jupiter.api.Test;

class ActivePlayerKillAnimationTest {

    @SuppressWarnings("JUnitTestMethodWithNoAssertions")
    @Test
    void playActivePlayerKillAnimation() {
        //given
        new GameStateMocks().mockTestPlayerList();
        //then
        new AnimationTester().testAnimation(new ActivePlayerKillAnimation());
    }
}

File: test/hud/animations/ActivePlayerKillAnimation.test.ts
import {describe, test} from 'vitest';
import {AnimationTester} from '../../AnimationTester.js';
import {GameStateMocks} from '../../GameStateMocks.js';
import {ActivePlayerKillAnimation} from '../../../src/hud/animations/ActivePlayerKillAnimation.js';

/**
 * Integration test for active player kill animation on actual hardware.
 */
describe('ActivePlayerKillAnimation Integration', () => {
    test('playActivePlayerKillAnimation', async () => {
        //given
        new GameStateMocks().playerList();
        //then
        await new AnimationTester().testAnimation(new ActivePlayerKillAnimation());
    }, 30000);
});
